docs/guide/middleware.md

```diff
@@ -0,0 +1,90 @@
+Middleware
+==========
+
+Middleware will be closure or class. Class must implemented by yii\web\MiddlewareInterface
+
+## Global
+
+Global middleware for all routes can add in application config:
+
+```php
+        [
+            'middleware' => \app\middleware\CopyrightMiddleware::class,
+            'priority' => 10,
+        ],        
+        function (\yii\web\Request $request, \yii\web\Response $response) {
+            $response->addHeader("heaader", "custom");
+        }
+```
+
+```php
+namespace app\middleware;
+
+use yii\web\MiddlewareInterface;
```

Comment:
https://github.com/php-fig/fig-standards/tree/master/proposed/http-handlers I think it makes sense to make it compatible with PSR interfaces (it's still not accepted but at least it's something commonly used).

Reply:
Need ServerRequestInterface. Now yii\web\Request implemented by RequestInterface.

Reply:
Then it should be applied.